src/ULog.ts

```diff
@@ -241,8 +243,19 @@ export class ULog {
       return;
     }
 
-    for (let i = range[0]; i <= range[1]; i++) {
-      const [_timestamp, offset, msgId] = timeIndex[i]!;
+    const startIndex = range[0];
+    const endIndex = range[1];
+
+    // The number of records between the start and end index. We will iterate this many times
```

Comment:
nit: `end - start` is not the number of records, it's the number of records minus 1

Reply:
The ranges are inclusive (from what I have learned)